spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaMessageChannelBinder.java

```diff
@@ -1483,6 +1502,10 @@ public void onSuccess(SendResult<K, V> result) {
 							KafkaMessageChannelBinder.this.logger
 									.debug("Sent to DLQ " + sb.toString());
 						}
+						if (!DlqSender.this.properties.getExtension().isAutoCommitOffset()) {
```

Comment:
Unfortunately, we need to get the actual `AckMode` from the listener container - the user might have reconfigured it to MANUAL_IMMEDIATE via the container customizer.